core/domain/customization_args_util.py

```diff
@@ -44,15 +44,23 @@ def get_full_customization_args(customization_args, ca_specs):
                     variable.
 
     Returns:
-        dict. The customization_args dict where missing keys are populated
-        with the default values.
+        tuple. A 2-tuple, the first being the customization_args dict where
+        missing keys are populated with the default values, and the second an
+        array of new content_id's from the default values.
     """
+    all_new_content_ids = []
+
     for ca_spec in ca_specs:
         if ca_spec.name not in customization_args:
-            customization_args[ca_spec.name] = {
-                'value': ca_spec.default_value
-            }
-    return customization_args
+            ca_value = {'value': ca_spec.default_value}
+            customization_args[ca_spec.name] = ca_value
+            new_content_ids = get_all_content_ids_in_cust_args(
+                {ca_spec.name: ca_value},
+                [ca_spec]
+            )
+            all_new_content_ids.extend(new_content_ids)
+
+    return customization_args, all_new_content_ids
```

Comment:
FYI this seems wrong conceptually. This function is supposed to do a particular thing but it seems like it's being co-opted to handle content IDs as well. Could a separate function handle that, so that each function has its own clear purpose?

Reply:
Redid the logic here and elsewhere.